design-proposals/multiarch-cluster.md

```diff
@@ -0,0 +1,154 @@
+
+**Virt-api**
+
+- VMI create admitter will now validate the architecture field provided in the VMI spec to determine if the multi-architecture feature gate has been enabled
+- VMI create admitter will retrieve the supported machine types using the added architecture field of the VMI spec
+- Validation of ARM64 specific settings will now rely on the architecture field of the VMI spec
+- VMI mutator’s setting of ARM64 specific settings will now rely on the architecture field of the VMI spec
+- VMI mutator will set the default architecture based on the cluster config item
+- VMI mutator will retrieve machine type based on architecture
+
+**Virt-controller**
+
+- Topology Hinter is no longer initialized with an architecture. Any methods that rely on architecture information will now be supplied from the VMI spec
+- Retrieval of the ovmfPath now relies on the architecture field of VMI spec instead of runtime.GOARCH
+
+**Virt-config**
+
+- Addition of MultiArchitecture feature gate 
+- Addition of defaults for each of the added KubevirtConfiguration fields
+ 
+**Virt-operator**
+
+- The addition of node-labeler initContainer for virt-handler daemonsets no longer performs architecture checks. The logic for determining whether the initContainer’s node-labeller.sh script runs has now been moved to the script itself.
+
+**Virt-launcher**
+
+- Given that the virt-launcher pod runs on a per VM basis, relying on runtime.GOARCH is an acceptable
+
+**Virt-handler**
+
+- Given that the virt-handler runs as a daemonset, relying on runtime.GOARCH is acceptable
+
+
+## API Examples
+
+**Example VM Spec**
+
+```yaml
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: mytestvm
+spec:
+  running: true
+  template:
+    spec:
+      architecture: arm64
+      domain:
+        devices:
+          disks:
+          - name: disk1
+            disk:
+              bus: virtio
+          interfaces:
+          - name: network1
+            bridge: {}
+      networks:
+              - name: network1
+                multus:
+                  networkName: mytestvm/mytestvm
+      volumes:
+           - name: disk1
+             emptyDisk:
+               capacity: 50G
+```
+
+**Example Kubevirt Configuration**
+
+```yaml	
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  certificateRotateStrategy: {}
+  configuration:
+    developerConfiguration:
+      featureGates: ["MultiArchitecture"]
+    archConfiguration:
+      arm64:
+        emulatedMachines: ["virt"]
+        ovmfPath: "/usr/share/AAVMF"
+      amd64:
+        emulatedMachines: ["pc-*"]
+        machineType: "q35"
+    defaultVMIArchitecture: "amd64"
```

Comment:
Copying this comment over from your design doc because it's important.  I wasn't clear to me from this doc that if `defaultVMIArchitecture` is unset the default will be `runtime.GOARCH`.   Can you add that somewhere?
```
Currently if you do not set the DefaultVMIArchitecture, it will default to
runtime.GOARCH. The same goes for all the arch specific defaults
(OVMF,Emulated Machines, etc), they will default to the hardcoded
X86,ARM defaults that already existed  as consts
```